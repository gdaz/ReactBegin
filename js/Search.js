import React from 'react'
import ShowCard from './ShowCard'

const {string, arrayOf, shape} = React.PropTypes

const Search = React.createClass({
  propTypes: {
    shows: arrayOf(shape({
      title: string.isRequired,
      description: string.isRequired
    }))
  },
  getInitialState () {
    return {
      msgSearch: ''
    }
  },
  handleSearchState (event) {
    this.setState({msgSearch: event.target.value})
  },
  render () {
    return (
      <div className='search'>
        <header>
          <h1>{this.state.msgSearch}</h1>
          <input value={this.state.msgSearch} onChange={this.handleSearchState} type='text' placeholder='Search' />
        </header>
        <div>
          {this.props.shows
          .filter((show) => {
            return `${show.title}`.toUpperCase()
            .indexOf(this.state.msgSearch.toUpperCase()) >= 0
          })
          .map((show) => {
            return (
              <ShowCard key={show.imdbID} {...show} />
            )
          })}
        </div>
      </div>
    )
  }
})

export default Search
