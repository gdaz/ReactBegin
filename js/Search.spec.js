import React from 'react'
import Search from './Seacrh'
import renderer from 'react-test-renderer'

test('Seacrh snapshot test', () => {
    const component = renderer.create(<Search />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
})