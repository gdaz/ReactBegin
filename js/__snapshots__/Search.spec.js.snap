// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Seacrh snapshot test 1`] = `
<div
  className="search"
>
  <header>
    <h1 />
    <input
      onChange={[Function]}
      placeholder="Search"
      type="text"
      value=""
    />
  </header>
  <div>
    <ShowCard
      description="A Congressman works with his equally conniving wife to exact revenge on the people who betrayed him."
      poster="hoc.jpg"
      title="House of Cards"
      year="2013–"
    />
    <ShowCard
      description="A blind lawyer with his other senses superhumanly enhanced fights crime as a costumed superhero."
      poster="dd.jpg"
      title="Daredevil"
      year="2015–"
    />
    <ShowCard
      description="The story of Piper Chapman, a woman in her thirties who is sentenced to fifteen months in prison after being convicted of a decade-old crime of transporting money for her drug-dealing girlfriend."
      poster="oitnb.jpg"
      title="Orange Is the New Black"
      year="2013–"
    />
    <ShowCard
      description="Level-headed son Michael Bluth takes over family affairs after his father is imprisoned. But the rest of his spoiled, dysfunctional family are making his job unbearable."
      poster="ad.jpg"
      title="Arrested Development"
      year="2003–"
    />
    <ShowCard
      description="A woman is rescued from a doomsday cult and starts life over again in New York City."
      poster="uks.jpg"
      title="Unbreakable Kimmy Schmidt"
      year="2015–"
    />
    <ShowCard
      description="In a world replete with greed, betrayal, sexual intrigue and rivalry, \\"Marco Polo\\" is based on the famed explorer's adventures in Kublai Khan's court in 13th century China."
      poster="mp.jpg"
      title="Marco Polo"
      year="2014–"
    />
    <ShowCard
      description="A family of adult siblings find that their secrets and scars are revealed when their black sheep of a brother returns home."
      poster="bl.jpg"
      title="Bloodline"
      year="2015–"
    />
    <ShowCard
      description="Meet the most beloved sitcom horse of the '90s - 20 years later. BoJack Horseman was the star of the hit TV show \\"Horsin' Around,\\" but today he's washed up, living in Hollywood, complaining about everything, and wearing colorful sweaters."
      poster="bh.jpg"
      title="BoJack Horseman"
      year="2014–"
    />
    <ShowCard
      description="A teenage girl is brutally murdered, sparking a hunt for her killer. But in a town where everyone hides a secret, will they find the monster among them?"
      poster="hg.jpg"
      title="Hemlock Grove"
      year="2013–"
    />
    <ShowCard
      description="A police investigation, the saga of a grieving family, and a Seattle mayoral campaign all interlock after the body of 17-year-old Rosie Larsen is found in the trunk of a submerged car."
      poster="tk.jpg"
      title="The Killing"
      year="2011–"
    />
    <ShowCard
      description="A program that follows a couple who must navigate the exhilarations and humiliations of intimacy, commitment and other things they were hoping to avoid."
      poster="l.jpg"
      title="Love"
      year="2016–"
    />
    <ShowCard
      description="The personal and professional life of Dev, a 30-year-old actor in New York."
      poster="mon.jpg"
      title="Master of None"
      year="2015–"
    />
    <ShowCard
      description="Will Arnett will star as a self-help guru who needs a decent amount of help himself."
      poster="f.jpg"
      title="Flaked"
      year="2016–"
    />
    <ShowCard
      description="A former superhero decides to reboot her life by becoming a private investigator."
      poster="jj.jpg"
      title="Jessica Jones"
      year="2015–"
    />
    <ShowCard
      description="A chronicled look at the criminal exploits of Colombian drug lord Pablo Escobar."
      poster="n.jpg"
      title="Narcos"
      year="2015–"
    />
    <ShowCard
      description="Follow the counselors and campers on their first day at Camp Firewood in the summer of 1981."
      poster="whas.jpg"
      title="Wet Hot American Summer: First Day of Camp"
      year="2015–"
    />
    <ShowCard
      description="When the patriarch of a prominent family dies, his heirs battle to determine who will gain control of his beloved soccer team: The Cuervos of Nuevo Toledo."
      poster="cdc.jpg"
      title="Club de Cuervos"
      year="2015–"
    />
    <ShowCard
      description="Filmed over a 10-year period, Making a Murderer is an unprecedented real-life thriller about Steven Avery, a DNA exoneree who, while in the midst of exposing corruption in local law enforcement, finds himself the prime suspect in a grisly new crime."
      poster="mam.jpg"
      title="Making a Murderer"
      year="2015–"
    />
    <ShowCard
      description="A group of people around the world are suddenly linked mentally, and must find a way to survive being hunted by those who see them as a threat to the world's order."
      poster="s8.jpg"
      title="Sense8"
      year="2015–"
    />
    <ShowCard
      description="D.J. Tanner-Fuller is a widow and mother of three. Things become too much to handle, so she asks for help from her sister Stephanie and her best friend Kimmy."
      poster="fh.jpg"
      title="Fuller House"
      year="2016–"
    />
    <ShowCard
      description="Bob and David emerge from the time machine they entered sixteen years ago and are reunited with the cast of Mr. Show."
      poster="wbd.jpg"
      title="w/ Bob & David"
      year="2016–"
    />
  </div>
</div>
`;
